### ISSUE TITLE
Vendoring documentation is outdated

### ISSUE THREAD

**Issue by oz123** [pypa/pipenv#4911]
### Issue description

The documentation included in `pipenv/vendor/README.md` is outdated.
I tried the command described there:
```
 ~/.local/bin/pipenv run inv
/home/oznt/.local/lib/python3.6/site-packages/pkg_resources/__init__.py:119: PkgResourcesDeprecationWarning: 1.6.5-ubuntu0.7-elementary19-ubuntu18.04.1 is an invalid version and will not be supported in a future release
  PkgResourcesDeprecationWarning,
Error: the command inv could not be found within PATH or Pipfile's [scripts].
```

### Expected result

I expect to be able to create and updated packages in the directory `vendor`.

### Actual result
The script failed with the error above.
Judging from the current CI, I think the correct way to do it is:
```
python -m invoke vendoring.update
```

Please run `$ pipenv --support`, and paste the results here. Don't put backticks (`` ` ``) around it! The output already contains Markdown formatting.

<details><summary>$ pipenv --support</summary>

Pipenv version: `'2022.1.8'`

Pipenv location: `'/home/oznt/.local/lib/python3.6/site-packages/pipenv'`

Python location: `'/usr/bin/python3'`

Python installations found:

  - `3.6.9`: `/usr/bin/python3.6`
  - `3.6.9`: `/usr/bin/python3`
  - `3.6.9`: `/usr/bin/python3.6m`
  - `2.7.17`: `/usr/bin/python`
  - `2.7.17`: `/usr/bin/python2`
  - `2.7.17`: `/usr/bin/python2.7`

PEP 508 Information:

```
{'implementation_name': 'cpython',
 'implementation_version': '3.6.9',
 'os_name': 'posix',
 'platform_machine': 'x86_64',
 'platform_python_implementation': 'CPython',
 'platform_release': '5.4.0-91-generic',
 'platform_system': 'Linux',
 'platform_version': '#102~18.04.1-Ubuntu SMP Thu Nov 11 14:46:36 UTC 2021',
 'python_full_version': '3.6.9',
 'python_version': '3.6',
 'sys_platform': 'linux'}
```

System environment variables:

  - `LS_COLORS`
  - `LC_MEASUREMENT`
  - `LESSCLOSE`
  - `LC_PAPER`
  - `LC_MONETARY`
  - `XDG_MENU_PREFIX`
  - `LANG`
  - `GDM_LANG`
  - `PANTHEON_TERMINAL_ID`
  - `DISPLAY`
  - `QT_STYLE_OVERRIDE`
  - `OLDPWD`
  - `GTK_CSD`
  - `COLORTERM`
  - `XDG_VTNR`
  - `GIO_LAUNCHED_DESKTOP_FILE_PID`
  - `GTK3_MODULES`
  - `SSH_AUTH_SOCK`
  - `LC_NAME`
  - `XDG_SESSION_ID`
  - `XDG_GREETER_DATA_DIR`
  - `USER`
  - `DESKTOP_SESSION`
  - `QT_QPA_PLATFORMTHEME`
  - `PWD`
  - `HOME`
  - `SSH_AGENT_PID`
  - `QT_ACCESSIBILITY`
  - `LIBVIRT_DEFAULT_URI`
  - `XDG_SESSION_TYPE`
  - `XDG_DATA_DIRS`
  - `XDG_SESSION_DESKTOP`
  - `LC_ADDRESS`
  - `LC_NUMERIC`
  - `GTK_MODULES`
  - `TERM`
  - `SHELL`
  - `VTE_VERSION`
  - `XDG_SEAT_PATH`
  - `QT_IM_MODULE`
  - `XMODIFIERS`
  - `XDG_CURRENT_DESKTOP`
  - `GPG_AGENT_INFO`
  - `GIO_LAUNCHED_DESKTOP_FILE`
  - `XDG_SEAT`
  - `SHLVL`
  - `LANGUAGE`
  - `PROMPT_COMMAND`
  - `LC_TELEPHONE`
  - `GDMSESSION`
  - `GNOME_DESKTOP_SESSION_ID`
  - `LOGNAME`
  - `DBUS_SESSION_BUS_ADDRESS`
  - `XDG_RUNTIME_DIR`
  - `XAUTHORITY`
  - `XDG_SESSION_PATH`
  - `XDG_CONFIG_DIRS`
  - `PATH`
  - `LC_IDENTIFICATION`
  - `SESSION_MANAGER`
  - `LESSOPEN`
  - `LC_TIME`
  - `_`
  - `PIP_SHIMS_BASE_MODULE`
  - `PIP_DISABLE_PIP_VERSION_CHECK`
  - `PYTHONDONTWRITEBYTECODE`
  - `PIP_PYTHON_PATH`
  - `PYTHONFINDER_IGNORE_UNSUPPORTED`

Pipenv–specific environment variables:

Debug–specific environment variables:

  - `PATH`: `/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/oznt/.rvm/bin:/home/oznt/.rvm/bin`
  - `SHELL`: `/bin/bash`
  - `LANG`: `en_US.UTF-8`
  - `PWD`: `/home/oznt/Software/pipenv`

---------------------------

Contents of `Pipfile` ('/home/oznt/Software/pipenv/Pipfile'):

```toml
[dev-packages]

flake8 = ">=3.3.0,<4"
pytest = "*"
mock = "*"
"delegator.py" = ">=0.0.13"
sphinx = "<=1.5.5"
"-e ." = "*"
toml = "*"

[packages]

pew = ">=0.1.26"
pep8 = "*"
pytest = "*"
pip-tools = "*"
semver = "*"
requests = "*"

[requires]
python_version = "3.6"

```
</details>

**Comment by Mause** [comment]
Pipenv does not have vendoring support built in, the documentation you read refers to the dependences that pipenv has included within itself - that is, vendored for it's own use

**Comment by oz123** [comment]
Yes, and that document I read is outdated. I'm am not expecting vendoring support. I'ml expecting the developer documentation to be updated.

**Comment by frostming** [comment]
> the command inv could not be found within PATH or Pipfile's [scripts].

`inv` is provides by `invoke` package, you should find it after run`pipenv install -d`

**Comment by oz123** [comment]
So, I need to install pipenv before I develop pipenv?
This should also be documented. Using `make install` from the makefile is broken (it tries the system's pip).

**Comment by frostming** [comment]
> So, I need to install pipenv before I develop pipenv?
That is the case, it can be install with `pip install -e .`

**Comment by oz123** [comment]
@frostming `pip install -e .` is broken.
See note: https://github.com/pypa/pipenv/blob/9770d6e11c4443742d1f4a617f1a8b4520ee2627/CONTRIBUTING.md?plain=1#L220

Currently, there is no clear way on how to start developing on pipenv.

**Comment by frostming** [comment]
Sorry, I didn't mention that you should do development in a vitualenv. That is the best practice.

**Comment by oz123** [comment]
So, the docs should say:
```
pip install --user pipenv
pipenv shell
pip install -e .
```

**Comment by willhoh** [comment]
As I just noticed, @oz123, you have changed the directories from vendor to docs/dev or reworked the whole documentation. Can this issueu be closed?

I would like to make a request to also describe how a proper development environment for pipenv should look like? Or at least recommend a good one? I have decided, after a lot of back and forth, to keep installing pipenv in a docker image and testing the function I have been working on. 

For example:
1. dockerfile:
```
FROM python:3.9

WORKDIR /opt

RUN apt update && apt upgrade -y && pip install --upgrade pip

COPY ./pipenv /opt/pipenv

CMD ["bash"]
```
2. `docker build -t pipenv_dev .`
3. `docker run -it --rm pipenv_dev bash`
4. `# pip install -e ./pipenv`
5. then follows teststs like `# pipenv install`
